Solver_interface/include/CGAL/Eigen_matrix.h

```diff
@@ -180,6 +180,36 @@ struct Eigen_sparse_matrix
       m_triplets.push_back(Triplet(i,j,val));
   }
 
+  /// Read access to a matrix coefficient.
+  ///
+  /// Complexity:
+  /// - O(log(n)) if the matrix is already built.
+  /// - O(n) if the matrix is not built.
+  /// n being the number of entries in the matrix.
+  ///
+  /// @commentheading Preconditions:
+  /// - 0 <= i < row_dimension().
+  /// - 0 <= j < column_dimension().
+  NT get_coef (unsigned int i, unsigned int j) const
+  {
+    CGAL_precondition(i < row_dimension());
+    CGAL_precondition(j < column_dimension());
+
+    NT val;
+    if (m_is_already_built)
+        val = m_matrix.coeffRef(i,j);
+    else
+    {
+      for(std::size_t t=0; t<m_triplets.size(); ++t)
+      {
+        if(m_triplets[t].col() == j &&
+           m_triplets[t].row() == i)
+          val = m_triplets[t].value();
```

Comment:
Actually, the vector `m_triplets` can contain multiple values for the same position (see `add_coef` function for example), so you have to accumulate, that is:
```
FT val = 0;
if(m_triplets[t].col() == j && triplets[t].row() == i)
  val += m_triplets[t].value();
```
This will also automatically handle a `0` return if there are no triplet for entry (i,j).

Lastly, you have to handle the possibility of the matrix being symmetric (see member `m_is_symmetric`). For symmetric matrices, `Eigen_sparse_matrix` stores only the lower triangle (`i>=j`). So if it is symmetric and `j > i`, you must swap them around.

Reply:
This was interesting, thanks!